dotCMS/src/main/java/com/dotcms/contenttype/business/ContentTypeAPIImpl.java

```diff
@@ -118,29 +118,45 @@ public ContentTypeAPIImpl(User user, boolean respectFrontendRoles) {
         APILocator.getPermissionAPI(), APILocator.getContentTypeFieldAPI(), APILocator.getLocalSystemEventsAPI());
   }
 
+  @Override
+  public void delete(final ContentType contentType) throws DotSecurityException, DotDataException {
+    deleteSync(contentType, true);
+  }
+
+  @Override
+  public void deleteSync(final ContentType contentType) throws DotSecurityException, DotDataException {
+    deleteSync(contentType, false);
+  }
+
   /**
-   * Content-Type Delete Entry point
-   * @param type Content Type that will be deleted
-   * @throws DotSecurityException
-   * @throws DotDataException
+   * Deletes the specified Content Type, either in the same database transaction or as a separate
+   * process.
+   *
+   * @param contentType The {@link ContentType} being deleted.
+   * @param async       If the deletion process should be executed asynchronously -- i.e.; in a
+   *                    separate process, set this to {@code true}.
+   *
+   * @throws DotSecurityException The specified User does not have edition permissions on this
+   *                              Content Type.
+   * @throws DotDataException     An error occurred when interacting with the database.
    */
-  @Override
-  public void delete(ContentType type) throws DotSecurityException, DotDataException {
-    if (!contentTypeCanBeDeleted(type)) {
-      Logger.warn(this, "Content Type " + type.name()
-              + " cannot be deleted because it is referenced by other content types");
+  private void deleteSync(final ContentType contentType, final boolean async) throws DotSecurityException, DotDataException {
```

Comment:
I think this method needs to be called just "delete" because the name "deleteSync" plus passing a flag for sync/async execution looks confusing

Reply:
Done!